master/buildbot/reporters/stash.py

```diff
@@ -83,3 +94,47 @@ def send(self, build):
                 content = yield response.content()
                 log.error("{code}: Unable to send Stash status: {content}",
                           code=response.code, content=content)
+
+
+class StashPRCommentPush(http.HttpStatusPushBase):
+    name = "StashPRCommentPush"
+
+    @defer.inlineCallbacks
+    def reconfigService(self, base_url, user, password, text=None,
+                        verbose=False, **kwargs):
+        yield http.HttpStatusPushBase.reconfigService(self, wantProperties=True,
+                                                      **kwargs)
+        self.text = text or Interpolate('Builder: %(prop:buildername)s '
+                                        'Status: %(prop:statustext)s')
+        self.verbose = verbose
+        self._http = yield httpclientservice.HTTPClientService.getService(
+            self.master, base_url, auth=(user, password),
+            debug=self.debug, verify=self.verify)
+
+    @defer.inlineCallbacks
+    def send(self, build):
+        if build['complete'] and build['properties'].has_key("pullrequesturl"):
+            yield self.sendPullRequestComment(build)
+
+    @defer.inlineCallbacks
+    def sendPullRequestComment(self, build):
+        props = Properties.fromDict(build['properties'])
+        pr_url = props.getProperty("pullrequesturl")
+        # we assume that the PR URL is well-formed as it comes from a PR event
+        path = urlparse(unicode2bytes(pr_url)).path
+        status = "SUCCESS" if build['results'] == SUCCESS else "FAILED"
+        props.setProperty('statustext', status, self.name)
+        props.setProperty('url', build['url'], self.name)
+        comment_text = yield props.render(self.text)
+        payload = {'text' : comment_text}
+        response = yield self._http.post(
+            STASH_COMMENT_API_URL.format(path=path),json=payload)
+
+        if response.code == 201:
```

Comment:
ok, I'll have to create another constant for the other class, maybe something like
BUILD_STATUS_ACCEPTED = 204
COMMENT_ACCEPTED = 201